libnetwork/netavark/config.go

```diff
@@ -125,6 +125,17 @@ func (n *netavarkNetwork) networkCreate(newNetwork *types.Network, defaultNet bo
 		if err != nil {
 			return nil, err
 		}
+	case types.WireGuardNetworkDriver:
+		// The network is purely created so that containers can join it
+		// with the actual interface being created via a config
+		// passed when running the container
+
+		// As no functionality is added via this network, subnets should not
+		// be created.
+		if newNetwork.Subnets != nil {
+			return nil, fmt.Errorf("unsupported network options for WireGuard driver: %s", "subnet")
+		}
```

Comment:
```suggestion
		if len(newNetwork.Subnets) != 0 {
			return nil, fmt.Errorf("subnets are not supported with wireguard driver")
		}
```
The problem is != nil will be true for an empty slice which is wrong, len() will return 0 even for nil pointers so this works for both cases.